File: RaftSwiftGRPC/Sources/RaftCore/Utils/Client/Put.swift
/// Client to Leader, used for adding or updating a key-value pair
public struct PutRequest: Sendable {
    /// Key to store
    public let key: String

    /// Optional value (absence implies a delete)
    public let value: String?

    /// Initializes a new PutRequest
    /// - Parameters:
    ///   - key: Key to store
    ///   - value: Optional value (absence implies a delete)
    public init(key: String, value: String?) {
        self.key = key
        self.value = value
    }
}

/// Leader to Client, response to PutRequest
public struct PutResponse: Sendable {
    /// Whether the operation succeeded
    public let success: Bool

    /// Optional leader hint if request failed
    public let leaderHint: Peer?

    /// Initializes a new PutResponse
    /// - Parameters:
    ///   - success: Whether the operation succeeded
    ///   - leaderHint: Optional leader hint if request failed
    public init(success: Bool, leaderHint: Peer? = nil) {
        self.success = success
        self.leaderHint = leaderHint
    }
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/utils/Client/Put.kt
package raft.utils.client

import raft.utils.types.Peer

/**
 * A request to put a key-value pair in the state machine.
 */
data class PutRequest(
    /**
     * Key to store
     */
    val key: String,
    /**
     * Optional value (absence implies a delete)
     */
    val value: String?
)

/**
 * A response to a put request.
 */
data class PutResponse(
    /**
     * Whether the operation succeeded
     */
    val success: Boolean,
    /**
     * Optional leader hint if request failed
     */
    val leaderHint: Peer? = null
)
